src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -518,7 +518,7 @@ private function createEncoder($config, ContainerBuilder $container)
         // Argon2i encoder
         if ('argon2i' === $config['algorithm']) {
             if (!Argon2iPasswordEncoder::isSupported()) {
-                throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please install the libsodium extension or upgrade to PHP 7.2+.');
+                throw new InvalidConfigurationException('Argon2i algorithm is not supported. For PHP < 7.2, install libsodium from PECL, for PHP >= 7.2 Please compile PHP with the --with-password-argon2 flag.');
```

Comment:
Checking the current php version to give the right recommendation would be better

Reply:
I guess it would, yeah.